LoopUI/Views/CGMStatusHUDView.swift

```diff
@@ -0,0 +1,172 @@
+        guard !statusStackView.arrangedSubviews.contains(alertStatusView) else {
+            return
+        }
+        
+        // need to also hide these view, since they will be added back to the stack at some point
+        glucoseValueHUD.isHidden = true
+        glucoseTrendHUD.isHidden = true
+        statusStackView.removeArrangedSubview(glucoseValueHUD)
+        statusStackView.removeArrangedSubview(glucoseTrendHUD)
+        
+        super.presentAlert()
+    }
+    
+    override public func dismissAlert() {
+        guard statusStackView.arrangedSubviews.contains(alertStatusView) else {
+            return
+        }
+        
+        super.dismissAlert()
+        
+        statusStackView.addArrangedSubview(glucoseValueHUD)
+        statusStackView.addArrangedSubview(glucoseTrendHUD)
+        glucoseValueHUD.isHidden = false
+        glucoseTrendHUD.isHidden = false
+    }
+    
+    public func setGlucoseQuantity(_ glucoseQuantity: Double, at glucoseStartDate: Date, unit: HKUnit, staleGlucoseAge: TimeInterval, sensor: SensorDisplayable?) {
+        var accessibilityStrings = [String]()
+        
+        let time = timeFormatter.string(from: glucoseStartDate)
+        caption?.text = time
+        
+        isStaleAt = glucoseStartDate.addingTimeInterval(staleGlucoseAge)
+        let glucoseValueCurrent = Date() < isStaleAt!
+        
+        let numberFormatter = NumberFormatter.glucoseFormatter(for: unit)
+        if let valueString = numberFormatter.string(from: glucoseQuantity) {
+            if glucoseValueCurrent {
+                glucoseValueHUD.glucoseLabel.text = valueString
+                startStalenessTimerIfNeeded()
+            } else {
+                glucoseValueHUD.glucoseLabel.text = GlucoseValueHUDView.staleGlucoseRepresentation
+            }
+            accessibilityStrings.append(String(format: LocalizedString("%1$@ at %2$@", comment: "Accessbility format value describing glucose: (1: glucose number)(2: glucose time)"), valueString, time))
+        }
+        
+        if let trend = sensor?.trendType, glucoseValueCurrent {
+            glucoseTrendHUD.setTrend(trend)
+            accessibilityStrings.append(trend.localizedDescription)
+        }
+        
+        if sensor == nil {
+            sensorAlertState = .missing
+        } else if sensor!.isStateValid == false {
+            sensorAlertState = .invalid
+            accessibilityStrings.append(LocalizedString("Needs attention", comment: "Accessibility label component for glucose HUD describing an invalid state"))
+        } else if sensor!.isLocal == false {
+            sensorAlertState = .remote
+        } else {
+            sensorAlertState = .ok
+        }
+        glucoseValueHUD.unitLabel.text = unit.localizedShortUnitString
+        accessibilityValue = accessibilityStrings.joined(separator: ", ")
+    }
+    
+    private lazy var timeFormatter = DateFormatter(timeStyle: .short)
+    
+    private var stalenessTimer: Timer?
```

Comment:
Seems to me `stalenessTimer`, and all of the associated variables/functions below should live in a `ViewModel`.

Reply:
I'll include a comment and handle this with the LOOP-1293, which is the CGM specific ticket for the status bar.